src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -101,6 +114,24 @@ protected PowerShellHost(string name, IRunspaceManager runspaceManager)
             InitializeSources();
 
             _sourceRepositoryProvider.PackageSourceProvider.PackageSourcesChanged += PackageSourceProvider_PackageSourcesChanged;
+            _restoreEvents.SolutionRestoreCompleted += RestoreEvents_SolutionRestoreCompleted;
+        }
+
+        private void RestoreEvents_SolutionRestoreCompleted(SolutionRestoredEventArgs args)
+        {
+            _latestSolutionSpecHash = args.SolutionSpecHash;
+
+            try
+            {
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+                {
+                    await ExecuteInitScriptsAsync();
```

Comment:
I'm a bit concerned with perf here. Loading up all the assets files could take some time and it looks like this would all be done on the UI thread.

This could be improved by either passing the packages to run init for from the assets file in the args. Or by loading these files off the UI thread in the background, then going to the UI thread.